src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -183,9 +183,13 @@ private bool IsOpenProperty(ODataProperty property)
 
             string propertyName = property.Name;
 
-            if (!this.JsonLightOutputContext.PropertyCacheHandler.InResourceSetScope())
+            if (this.JsonLightOutputContext.MessageWriterSettings.Validations != ValidationKinds.None)

```

Comment:
maybe:
```C#
if ( ... .InResourceSetScope)
{
          if ( .. != ValidationKinds.None)
          { 
                    ValidatePropertyName(...);
          }
          ...
}
else
{
      ...
}
```

Reply:
In the else case, PropertyCacheHandler eventually also called ValidatePropertyName. So it was called in both cases, but in weird places. Moving it out of the if statement provided a common place to validate, regardless of whether or not we were in a resourcesetscope (which shouldn't affect property validation).

---
In reply to: [154829506](https://github.com/OData/odata.net/pull/1010#discussion_r154829506) [](ancestors = 154829506)